src/main/kotlin/app/revanced/patches/twitter/misc/fxtwitter/FxTwitterPatch.kt

```diff
@@ -0,0 +1,134 @@
+package app.revanced.patches.twitter.misc.fxtwitter
+
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.InstructionExtensions.addInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.getInstructions
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchException
+import app.revanced.patcher.patch.annotation.CompatiblePackage
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patcher.patch.options.types.StringPatchOption.Companion.stringPatchOption
+import app.revanced.patches.twitter.misc.fxtwitter.fingerprints.FxTwitterPatchFingerprint
+import app.revanced.patches.twitter.misc.fxtwitter.fingerprints.FxTwitterPatchFingerprintResource
+import com.android.tools.smali.dexlib2.Opcode
+import com.android.tools.smali.dexlib2.builder.BuilderInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.ReferenceInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.formats.Instruction35c
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+import com.android.tools.smali.dexlib2.iface.reference.StringReference
+import java.util.logging.Logger
+
+
+@Patch(
+    name = "FxTwitter",
+    description = "Replaces the default Twitter share link with FxTwitter.",
+    compatiblePackages = [CompatiblePackage("com.twitter.android")]
+)
+@Suppress("unused")
+object FxTwitterPatch : BytecodePatch(
+    setOf(FxTwitterPatchFingerprint, FxTwitterPatchFingerprintResource)
+) {
+    private var shareLinkBase by stringPatchOption(
+        key = "twitter_share_link_base",
+        default = "fxtwitter.com",
+        title = "Embed builder domain",
+        description = "The embed builder domain used to share the twitter links. (Default: fxtwitter.com) known alts: vxtwitter.com, fixupx.com.",
+    )
+
+    override fun execute(context: BytecodeContext) {
+        var twitterGetShareLinkMethod = FxTwitterPatchFingerprint.result?.apply {
+            val instructions = this.mutableMethod.getInstructions()
+
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.CONST_STRING) {
+                    val string = ((instruction as ReferenceInstruction).reference as StringReference).string
+
+                    if (string.contains("https://twitter.com/%1\$s/status/%2\$d") || string.contains("https://x.com/%1\$s/status/%2\$d")) {
+                        val overrideRegister = (instruction as OneRegisterInstruction).registerA
```

Comment:
Why not instead use `FxTwitterPatchFingerprint.result!!.stringScanResult`?

Reply:
I'm not really sure what to do here, could you explain what you mean by that? I don't see stringScanResult anywhere

Reply:
Alrigh, got it, seems to work